contracts/src/v0.8/dev/VRFSubscriptionBalanceMonitor.sol

```diff
@@ -0,0 +1,314 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.6;
+
+import "../ConfirmedOwner.sol";
+import "../interfaces/KeeperCompatibleInterface.sol";
+import "../interfaces/VRFCoordinatorV2Interface.sol";
+import "../interfaces/LinkTokenInterface.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+
+/**
+ * @title The VRFSubscriptionBalanceMonitor contract.
+ * @notice A keeper-compatible contract that monitors and funds VRF subscriptions.
+ */
+contract VRFSubscriptionBalanceMonitor is ConfirmedOwner, Pausable, KeeperCompatibleInterface {
+  VRFCoordinatorV2Interface public COORDINATOR;
+  LinkTokenInterface public LINKTOKEN;
```

Comment:
Do these have to be mutable? We'd have a smaller public ABI if they're `immutable` and remove the setter functions. I'm not sure when we'd change these (certainly not the LINK token, probably not the VRF coordinator).

Reply:
Other balance monitor contracts allow for these values to be changed. Not sure why we'd want that but it matches the current flow.